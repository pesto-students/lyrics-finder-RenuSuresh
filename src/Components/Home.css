.home__row {
  display: flex;
  overflow-y: hidden;
  overflow-x: scroll;
  padding: 20px;
}
.home__row::-webkit-scrollbar {
  display: none;
}
.home__card {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  margin: auto;
  text-align: center;
  display: flex;
  flex-direction: column;
  background-color: #fff;
  width: 200px;
  transition: transform 450ms;
  margin: 10px;
}

.home__card:hover {
  transform: scale(1.08);
  opacity: 1;
}
.home__card > img {
  height: 135px;
}
