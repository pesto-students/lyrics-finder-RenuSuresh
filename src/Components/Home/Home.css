.home__row {
  display: flex;
  overflow-y: hidden;
  overflow-x: scroll;
  padding: 20px;
}
.home__row::-webkit-scrollbar {
  display: none;
}
.home__card {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  margin: auto;
  text-align: center;
  display: flex;
  flex-direction: column;
  background-color: #fff;
  width: 200px;
  transition: transform 450ms;
  margin: 10px;
  padding: 7px;
}

.home__card:hover {
  transform: scale(1.08);
  opacity: 1;
}
.home__card > img {
  height: 135px;
}

label {
  font-family: "Abel";
  font-size: 24px;
  font-weight: 600;
  color: #860264;
}
