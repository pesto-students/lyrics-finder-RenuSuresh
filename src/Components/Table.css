.table__background {
  height: fit-content;
  background-color: #360000;
  background-image: linear-gradient(to right, #d88b17, #6e6726);
}
.table__div {
  display: flex;
  justify-content: center;
}
p {
  display: flex;
  justify-content: center;
  font-size: 40px;
  font-weight: 600;
  font-family: cursive;
  text-shadow: 0 0 14px #d88b17;
}
.table__table {
  display: flex;
  justify-content: center;
}
table {
  border: 1px solid #ccc;
  border-collapse: collapse;
  padding: 0;
  /* width: 60%; */
  table-layout: fixed;
  margin: 0 auto;
  min-width: fit-content;
  margin: 25px;
  /* box-shadow: 0 4px 8px 0 red, 0 4px 8px 0 red; */
  box-shadow: 4px 8px 0 rgba(0, 0, 0, 0), 0 6px 20px 0 rgba(237, 172, 141, 0.8);
}

table caption {
  font-size: 1.5em;
  margin: 0.5em 0 0.75em;
}

table tr {
  background-color: #f8f8f8;
  border: 1px solid #ddd;
  padding: 0.35em;
}

table th,
table td {
  padding: 0.625em;
}

table th {
  font-size: 0.85em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
}

@media screen and (max-width: 600px) {
  table {
    border: 0;
  }

  table caption {
    font-size: 1.3em;
  }

  table thead {
    border: none;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px;
  }

  table tr {
    border: 3px solid #ddd;
    display: block;
    margin-bottom: 0.625em;
  }

  table td {
    border: 1px solid #ddd;
    display: block;
    font-size: 0.8em;
    text-align: left;
  }

  table td::before {
    content: attr(data-label);
    float: left;
    font-weight: bold;
    text-transform: uppercase;
  }

  table td:last-child {
    border-bottom: 0;
  }
}

tbody tr:hover {
  background-color: #d6d6d6;
}
.table__info {
  display: flex;
  flex-direction: column;
}
.table__title {
  font-size: small;
}
.table__album {
  font-family: "Courier New", Courier, monospace;
  font-size: 17px;
  font-weight: 600;
}
.table__img__col {
  width: 15%;
}
.table__caption {
  width: auto;
  display: flex;
  justify-content: center;
  margin-top: 64px;
}
.table__caption > p {
  color: #eeeaf0;
  border-bottom: 3px double #ffffff;
}
.table__div__border {
  border: 5px double white;
  width: 60%;
}
.close {
  color: white;
  font-size: xxx-large;
  margin-right: 15px;
  opacity: 1;
}
.close:hover {
  color: #ffc766;
  outline: none;
}
