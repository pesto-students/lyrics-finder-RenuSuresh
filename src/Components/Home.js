import React, { useState } from "react";
import "./Home.css";
import { homeTiles } from "../properties";
import Table from "./Table";

function Home() {
  const [musicType, setMusicType] = useState("");
  const handleClick = (music) => {
    console.log("music type is >>>>", music);
    setMusicType(music.type);
  };
  return (
    <div className="home">
      <div className="home__row">
        {homeTiles.map((music) => (
          <div
            className="home__card"
            onClick={() => handleClick(music)}
            key={music.id}
          >
            <img src={music.image} alt={music.type} />
            <h1>{music.type}</h1>
          </div>
        ))}
      </div>
      {musicType && <Table musicType={musicType} />}
    </div>
  );
}

export default Home;
