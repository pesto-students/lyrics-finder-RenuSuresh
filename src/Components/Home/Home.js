import React, { useState } from "react";
import "./Home.css";
import { homeTiles } from "../../services/constant/properties";
import Table from "../Table/Table";

function Home() {
  const [musicType, setMusicType] = useState("");
  const handleClick = (music) => {
    setMusicType(music.type);
  };
  const hideTable = () => {
    setMusicType("");
  };
  return (
    <div className="home">
      <div className="home__row">
        {homeTiles.map((music) => (
          <div
            className="home__card"
            onClick={() => handleClick(music)}
            key={music.id}
            data-testid={`card-${music.id}`}
          >
            <img src={music.image} alt={music.type} />
            <label>{music.type}</label>
          </div>
        ))}
      </div>

      {musicType && <Table musicType={musicType} hideTable={hideTable} />}
    </div>
  );
}

export default Home;
