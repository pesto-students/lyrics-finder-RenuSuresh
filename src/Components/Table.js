import React, { useEffect, useState } from "react";
import "./Table.css";
import { getSuggest } from "../axios";
import LyricsModal from "./LyricsModal";
import Pagination from "./Pagination";
import BeatLoader from "react-spinners/BeatLoader";

function TableData({ currentGenre, getLyrics }) {
  return (
    <table>
      <tbody>
        {currentGenre.map((Element) => (
          <tr
            key={Element.id}
            onClick={() =>
              getLyrics(
                Element.artist.name,
                Element.title,
                Element.album.title,
                Element.artist.picture_medium
              )
            }
          >
            <td className="table__img__col">
              <img src={Element.album.cover_small} alt="album" />
            </td>
            <td className="table__info">
              <span className="table__album">{Element.album.title}</span>
              <span className="table__title">{Element.title}</span>
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
}

function Table({ musicType, hideTable }) {
  const [genre, setGenre] = useState([]);
  const [artist, setArtist] = useState("");
  const [title, setTitle] = useState("");
  const [currentPage, setCurrentPage] = useState(1);
  const [genrePerPage] = useState(5);
  const [loading, setLoading] = useState(true);
  const [image, setImage] = useState("");
  const [albumTitle, setAlbumTitle] = useState("");

  // get current data
  const indexOfLastGenre = currentPage * genrePerPage;
  const indexOfFirstGenre = indexOfLastGenre - genrePerPage;
  const currentGenre = genre.slice(indexOfFirstGenre, indexOfLastGenre);

  // change page
  const paginate = (pageNumber) => setCurrentPage(pageNumber);

  useEffect(() => {
    async function fetchData() {
      const request = await getSuggest.get(musicType);

      setGenre(request.data.data);
      setLoading(false);
      return request;
    }
    fetchData();
  }, [musicType]);
  console.log("genre>>>", genre);
  const getLyrics = (artist, title, album_title, image) => {
    setTitle(title);
    setArtist(artist);
    setImage(image);
    setAlbumTitle(album_title);
    console.log("albumtitle>>>>", album_title);
  };

  return (
    <div className="table__div">
      <div className="table__div__border">
        <button type="button" className="close" onClick={hideTable}>
          &times;
        </button>
        <div className="table__caption">
          <p>{musicType}</p>
        </div>

        <div className="table__table">
          {loading ? (
            <BeatLoader loading={loading} color={"#ffffff"} />
          ) : (
            <TableData currentGenre={currentGenre} getLyrics={getLyrics} />
          )}
        </div>
        <Pagination
          genrePerPage={genrePerPage}
          totalGenre={genre.length}
          paginate={paginate}
        />
        {artist && (
          <LyricsModal
            artist={artist}
            title={title}
            image={image}
            album_title={albumTitle}
          />
        )}
      </div>
    </div>
  );
}

export default Table;
