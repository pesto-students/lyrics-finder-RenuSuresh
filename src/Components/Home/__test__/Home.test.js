import { configure, EnzymeAdapter } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { shallow } from "enzyme";
import Home from "../Home";
import Enzyme from "enzyme";
import {
  render,
  fireEvent,
  getByTestId,
  getByText,
} from "@testing-library/react";
import toJson from "enzyme-to-json";
import Table from "../../Table/Table";
configure({ adapter: new Adapter() });
const { mount } = Enzyme;

describe("Home", () => {
  test("snapshot <Home/>", () => {
    const homeTree = mount(<Home />);
    expect(toJson(homeTree)).toMatchSnapshot();
  });
  test("card to be clickable", () => {
    const { container } = render(<Home />);
    const tiles = getByTestId(container, "card-1");
    expect(fireEvent.click(tiles)).toBeTruthy();
  });
  test("on click get the table", () => {
    const { container } = render(<Home />);
    const tiles = getByTestId(container, "card-1");
    fireEvent.click(tiles);
    const homeWrapper = shallow(<Home />);

    expect(homeWrapper.find(Table)).toBeDefined();
  });
});
