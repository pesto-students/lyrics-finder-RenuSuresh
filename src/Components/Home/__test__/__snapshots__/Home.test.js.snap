// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot <Home/> 1`] = `
<Home>
  <div
    className="home"
  >
    <div
      className="home__row"
    >
      <div
        className="home__card"
        data-testid="card-1"
        key="1"
        onClick={[Function]}
      >
        <img
          alt="Popular Music"
          src="https://music.usc.edu/files/2016/07/PopularMusic_10_Header.jpg"
        />
        <label>
          Popular Music
        </label>
      </div>
      <div
        className="home__card"
        data-testid="card-2"
        key="2"
        onClick={[Function]}
      >
        <img
          alt="Rock"
          src="https://akm-img-a-in.tosshub.com/indiatoday/music-day-story,-facebook_647_062116084641.jpg?wqTCLtMqwQpeuevah3aCZLvcopVtZz24&size=770:433"
        />
        <label>
          Rock
        </label>
      </div>
      <div
        className="home__card"
        data-testid="card-3"
        key="3"
        onClick={[Function]}
      >
        <img
          alt="Classical"
          src="https://upload.wikimedia.org/wikipedia/commons/0/0f/Sawai.jpg"
        />
        <label>
          Classical
        </label>
      </div>
      <div
        className="home__card"
        data-testid="card-4"
        key="4"
        onClick={[Function]}
      >
        <img
          alt="Soul music"
          src="https://mediad.publicbroadcasting.net/p/shared/npr/styles/x_large/nprshared/201902/698639816.jpg"
        />
        <label>
          Soul music
        </label>
      </div>
      <div
        className="home__card"
        data-testid="card-5"
        key="5"
        onClick={[Function]}
      >
        <img
          alt="Country music"
          src="https://www.cmuse.org/wp-content/uploads/2020/06/Characteristics-of-Country-Music.jpg"
        />
        <label>
          Country music
        </label>
      </div>
      <div
        className="home__card"
        data-testid="card-6"
        key="6"
        onClick={[Function]}
      >
        <img
          alt="Orchestra"
          src="https://www.grammy.com/sites/com/files/styles/twitter_image/public/chicagosymphonyorchestra_toddrosenberg.jpg?itok=lgnURVxN"
        />
        <label>
          Orchestra
        </label>
      </div>
      <div
        className="home__card"
        data-testid="card-7"
        key="7"
        onClick={[Function]}
      >
        <img
          alt="Folk Music"
          src="https://www.liveabout.com/thmb/GTjEPzMYKvwiEcU25JZeonzKYwA=/1500x1000/filters:no_upscale():max_bytes(150000):strip_icc()/TheAlmanacSingers-5902520f3df78c5456c436a1.jpg"
        />
        <label>
          Folk Music
        </label>
      </div>
      <div
        className="home__card"
        data-testid="card-8"
        key="8"
        onClick={[Function]}
      >
        <img
          alt="Heavy Metal"
          src="https://media.npr.org/assets/img/2019/07/15/murder-falcon_wide-6a7466500034276836ba06a964a01981181c0817-s800-c85.jpg"
        />
        <label>
          Heavy Metal
        </label>
      </div>
    </div>
  </div>
</Home>
`;
